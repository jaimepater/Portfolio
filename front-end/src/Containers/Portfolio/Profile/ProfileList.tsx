import * as React from 'react';
import { useEffect, useState } from 'react';
import ProfileItem from './ProfileItem';

interface IProfileListProps {
  profiles: Array<string>;
  time: number;
}

const ProfileList = ({ profiles, time }: IProfileListProps) => {
  const [profile, setProfile] = useState({ id: 0, name: '' });
  const showProfile = (index: any) => {
    if (!isNaN(index)) {
      setProfile({ name: profiles[index], id: index });
      setTimeout(showProfile.bind(undefined, (index + 1) % profiles.length), time);
    }
  };

  useEffect(() => {
    showProfile(0);
  }, [profiles, profiles.length, time]);

  return <ProfileItem key={profile.id} profile={profile.name} time={time} />;
};

export default ProfileList;
