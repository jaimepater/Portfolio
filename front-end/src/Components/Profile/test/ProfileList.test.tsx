import React from "react";
import { shallow} from 'enzyme';
import ProfileList from "../ProfileList";
import portfolioData from "../../../../Commons/__mocks__/Stores/profileData.json";
const  list = portfolioData.profile.list


describe('ProfileName', () => {
  it('Renders ProfileName', () => {
    const shallowComponent = shallow(<ProfileList profiles={list} time={5000} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});