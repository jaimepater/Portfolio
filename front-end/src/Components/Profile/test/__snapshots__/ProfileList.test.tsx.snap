// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileName Renders ProfileName 1`] = `
<ProfileItem
  key="0"
  profile=""
  time={5000}
/>
`;
